src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs

```diff
@@ -144,20 +148,17 @@ private bool AnyTopLevelTypeMatchesDocumentName(State state, CancellationToken c
         protected bool TypeMatchesDocumentName(
             TTypeDeclarationSyntax typeNode,
             string typeName,
-            string documentName,
+            string documentNameWithoutExtension,
             SemanticModel semanticModel,
             CancellationToken cancellationToken)
         {
-            // trim extensions, if present.
-            documentName = Path.GetFileNameWithoutExtension(documentName);
```

Comment:
Correct.  And so, trimming again would cause problems.  i.e. if you had "Outer.Inner.cs" we would have already trimmed off ".cs" leaving "Outer.Inner", this extra call would then trim that again to "Outer".